From 681ef40f95a82e79637aad668a2c38851c8df9de Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Thu, 18 Jun 2020 17:40:59 +0530
Subject: [PATCH] glib: Never use MessageBox on UWP apps

It used to be allowed when building in debug mode, but it isn't
allowed at all anymore.
---
 glib/gbacktrace.c | 5 ++---
 glib/gmessages.c  | 5 ++---
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/glib/gbacktrace.c b/glib/gbacktrace.c
index 86f8841..98db3eb 100644
--- a/glib/gbacktrace.c
+++ b/glib/gbacktrace.c
@@ -192,9 +192,8 @@ g_on_error_query (const gchar *prg_name)
   if (!prg_name)
     prg_name = g_get_prgname ();
 
-  /* MessageBox is allowed on UWP apps only when building against
-   * the debug CRT, which will set -D_DEBUG */
-#if defined(_DEBUG) || !defined(G_WINAPI_ONLY_APP)
+  /* MessageBox is not allowed on UWP apps */
+#if !defined(G_WINAPI_ONLY_APP)
   MessageBox (NULL, "g_on_error_query called, program terminating",
               (prg_name && *prg_name) ? prg_name : NULL,
               MB_OK|MB_ICONERROR);
diff --git a/glib/gmessages.c b/glib/gmessages.c
index 201192f..3271553 100644
--- a/glib/gmessages.c
+++ b/glib/gmessages.c
@@ -1360,9 +1360,8 @@ g_logv (const gchar   *log_domain,
 
           if ((test_level & G_LOG_FLAG_FATAL) && !masquerade_fatal)
             {
-              /* MessageBox is allowed on UWP apps only when building against
-               * the debug CRT, which will set -D_DEBUG */
-#if defined(G_OS_WIN32) && (defined(_DEBUG) || !defined(G_WINAPI_ONLY_APP))
+              /* MessageBox is not allowed on UWP apps */
+#if defined(G_OS_WIN32) && !defined(G_WINAPI_ONLY_APP)
               if (win32_keep_fatal_message)
                 {
                   gchar *locale_msg = g_locale_from_utf8 (fatal_msg_buf, -1, NULL, NULL, NULL);
@@ -2684,7 +2684,7 @@ handled:
     {
       /* MessageBox is allowed on UWP apps only when building against
        * the debug CRT, which will set -D_DEBUG */
-#if defined(G_OS_WIN32) && (defined(_DEBUG) || !defined(G_WINAPI_ONLY_APP))
+#if defined(G_OS_WIN32) && !defined(G_WINAPI_ONLY_APP)
       if (!g_test_initialized ())
         {
           gchar *locale_msg = NULL;
-- 
2.27.0.windows.1

